include/ignition/math/SpeedLimiter.hh

```diff
@@ -55,77 +33,100 @@ inline namespace IGNITION_MATH_VERSION_NAMESPACE {
-        double _minAcceleration       = -INF_D,
-        double _maxAcceleration       = INF_D,
-        double _minJerk               = -INF_D,
-        double _maxJerk               = INF_D);
+    /// There are no limits by default.
+    public: SpeedLimiter();
 
     /// \brief Destructor.
     public: ~SpeedLimiter();
 
-    /// \brief Limit velocity, acceleration and jerk, as enabled during
-    /// construction.
-    /// \param [in, out] _v Velocity to limit [m/s].
-    /// \param [in] _v0 Previous velocity to v  [m/s].
-    /// \param [in] _v1 Previous velocity to v0 [m/s].
+    /// \brief Set minimum velocity limit in m/s, usually <= 0.
+    /// \param[in] _lim Minimum velocity.
+    public: void SetMinVelocity(double _lim);
+
+    /// \brief Get minimum velocity limit, defaults to negative infinity.
+    /// \return Minimum velocity.
+    public: double MinVelocity() const;
+
+    /// \brief Set maximum velocity limit in m/s, usually >= 0.
+    /// \param[in] _lim Maximum velocity.
+    public: void SetMaxVelocity(double _lim);
+
+    /// \brief Get maximum velocity limit, defaults to positive infinity.
+    /// \return Maximum velocity.
+    public: double MaxVelocity() const;
+
+    /// \brief Set minimum acceleration limit in m/s, usually <= 0.
+    /// \param[in] _lim Minimum acceleration.
+    public: void SetMinAcceleration(double _lim);
+
+    /// \brief Get minimum acceleration limit, defaults to negative infinity.
+    /// \return Minimum acceleration.
+    public: double MinAcceleration() const;
+
+    /// \brief Set maximum acceleration limit in m/s, usually >= 0.
+    /// \param[in] _lim Maximum acceleration.
+    public: void SetMaxAcceleration(double _lim);
+
+    /// \brief Get maximum acceleration limit, defaults to positive infinity.
+    /// \return Maximum acceleration.
+    public: double MaxAcceleration() const;
+
+    /// \brief Set minimum jerk limit in m/s, usually <= 0.
+    /// \param[in] _lim Minimum jerk.
+    public: void SetMinJerk(double _lim);
+
+    /// \brief Get minimum jerk limit, defaults to negative infinity.
+    /// \return Minimum jerk.
+    public: double MinJerk() const;
+
+    /// \brief Set maximum jerk limit in m/s, usually >= 0.
+    /// \param[in] _lim Maximum jerk.
+    public: void SetMaxJerk(double _lim);
+
+    /// \brief Get maximum jerk limit, defaults to positive infinity.
+    /// \return Maximum jerk.
+    public: double MaxJerk() const;
+
+    /// \brief Limit velocity, acceleration and jerk.
+    /// \param [in, out] _vel Velocity to limit [m/s].
+    /// \param [in] _prevVel Previous velocity to v  [m/s].
+    /// \param [in] _prevPrevVel Previous velocity to prevVel [m/s].
     /// \param [in] _dt Time step.
-    /// \return Limiting factor, which is (out_v / in_v).
-    public: double Limit(double &_v,
-                         double _v0,
-                         double _v1,
+    /// \return Limiting factor, which is (out _vel - in _vel).
```

Comment:
I wouldn't call it a limiting factor any more since it's not a ratio; I would call it the difference between the output and input values of `_vel` (output - input)